backend/RestAPI/Controllers/VideosController.cs

```diff
@@ -341,6 +341,47 @@ public async Task<ActionResult<VideoDto>> ChangeTitle(Guid id, string newTitle)
             return Ok(response);
         }
 
+        [HttpPatch, Route("MarkDeleted")]
+        public async Task<ActionResult> MarkVideosForDeletion([FromBody] List<Guid> ids)
+        {
+            var userIdClaim = User.FindFirst(x => x.Type == ClaimTypes.NameIdentifier);
+            if (userIdClaim is null)
+            {
+                return NotFound();
+            }
+
+            if (!Guid.TryParse(userIdClaim.Value, out var userId))
+            {
+                return NotFound();
+            }
+
+            var user = await _usersRepository.GetUserById(userId);
+            if (user is null)
+            {
+                return NotFound();
+            }
+
+            foreach(Guid id in ids)
+            {
+                var video = await _videosRepository.GetVideoById(id);
+                if (video == null)
+                {
+                    return NotFound();
+                }
+
+                if (video.UserId != user.Id)
+                {
+                    return NotFound();
+                }
+
+                video.DeleteDate = DateTime.Today.AddMonths(1);
```

Comment:
Why are you using `AddMonths` here?

Reply:
Also, shouldn't this be moved to Service class?

Reply:
![image](https://user-images.githubusercontent.com/29150754/118652136-f2e31580-b7ee-11eb-988c-1e51cb02326b.png)
You could use this, I think